p2p/src/authenticated/actors/tracker/record.rs

```diff
@@ -138,95 +146,156 @@ impl<C: Verifier> Record<C> {
+    /// checked against the existing record to ensure that we correctly attribute the failure.
+    pub fn dial_failure(&mut self, socket: SocketAddr) {
+        if let Address::Discovered(info, fails) = &mut self.address {
+            if info.socket == socket {
+                *fails += 1;
+            }
         }
     }
 
-    /// Returns `true` if we want to attempt to refresh the peer information if not connected to
-    /// this peer.
+    // ---------- Getters ----------
+
+    /// Returns `true` if the record is blocked.
+    pub fn blocked(&self) -> bool {
+        matches!(self.address, Address::Blocked)
+    }
+
+    /// Returns `true` if the record is dialable.
     ///
-    /// `false` only for `Myself` and `Blocked` peers.
-    pub fn refreshable(&self) -> bool {
-        match self.address {
-            Address::Myself(_) | Address::Blocked => false,
-            Address::Unknown
-            | Address::Bootstrapper(_)
-            | Address::Discovered(_)
-            | Address::Persistent(_) => true,
-        }
+    /// A record is dialable if:
+    /// - We have the socket address of the peer
+    /// - It is not ourselves
+    /// - We are not already connected
+    pub fn dialable(&self) -> bool {
+        matches!(self.status, Status::Inert | Status::Reserved)
+            && matches!(
+                self.address,
+                Address::Bootstrapper(_) | Address::Discovered(_, _)
+            )
     }
 
-    /// Get the address of the peer if known.
-    pub fn address(&self) -> Option<SocketAddr> {
+    /// Return the socket of the peer, if known.
+    pub fn socket(&self) -> Option<SocketAddr> {
         match &self.address {
             Address::Unknown => None,
-            Address::Myself(peer_info) => Some(peer_info.socket),
+            Address::Myself(info) => Some(info.socket),
             Address::Bootstrapper(socket) => Some(*socket),
-            Address::Discovered(peer_info) => Some(peer_info.socket),
-            Address::Persistent(peer_info) => Some(peer_info.socket),
+            Address::Discovered(info, _) => Some(info.socket),
             Address::Blocked => None,
         }
     }
 
-    /// Get the peer information if known.
-    pub fn peer_info(&self) -> Option<&PeerInfo<C>> {
+    /// Get the peer information if it is sharable. The information is considered sharable if it is
+    /// known and we are connected to the peer.
+    pub fn sharable_info(&self) -> Option<PeerInfo<C>> {
         match &self.address {
             Address::Unknown => None,
-            Address::Myself(peer_info) => Some(peer_info),
+            Address::Myself(info) => Some(info),
             Address::Bootstrapper(_) => None,
-            Address::Discovered(peer_info) => Some(peer_info),
-            Address::Persistent(peer_info) => Some(peer_info),
+            Address::Discovered(info, _) => (self.status == Status::Active).then_some(info),
```

Comment:
Related to my other comment around only "resetting if dialed successfully", I'm wondering if we should only make sharable if we dialed?

I suppose this could really slowdown discovery if someone dials everyone before any have a chance to dial them?

Reply:
No good answer that I can come up with right now. I created an issue [here](https://github.com/commonwarexyz/monorepo/issues/927) for further discussion

Reply:
I think the "best" approach is to do what the PR does now (share anything for anyone we are connected to, even if it might not be active).